File: java/design-patterns/memento/originator/TextEditor.java
public class TextEditor {
    private String text = "";
    public void type(String value) { this.text += value; }
    public String getText() { return text; }

    public Memento save() { return new Memento(text); }
    public void restore(Memento m) {
        this.text = m.getState();
    }
}

File: python/design-patterns/memento/originator/text_editor.py
class TextEditor:
    def __init__(self):
        self._text = ""
    def type(self, val): self._text += val
    def get_text(self): return self._text
    
    def save(self) -> Memento:
        return Memento(self._text)
    def restore(self, memento: Memento):
        self._text = memento.get_state()